Core/clim-core/table-formatting.lisp

```diff
@@ -259,33 +312,18 @@ skips intervening non-table output record structures."))
                 (replay-output-record record stream region x-offset y-offset))
             (nreverse other-records)))))
 
-(defmacro formatting-table ((&optional (stream t)
-                                       &rest args
-                                       &key x-spacing y-spacing
-                                       multiple-columns
-                                       multiple-columns-x-spacing
-                                       equalize-column-widths (move-cursor t)
-                                       (record-type ''standard-table-output-record)
-                                       &allow-other-keys)
-                            &body body)
-  (declare (ignore x-spacing y-spacing multiple-columns
-                   multiple-columns-x-spacing
-                   equalize-column-widths move-cursor record-type))
-  (gen-invoke-trampoline 'invoke-formatting-table
-                         (list (stream-designator-symbol stream
-                                                         '*standard-output*))
-                         args
-                         body))
+(eval-when (:compile-toplevel :load-toplevel :execute)
+  (alexandria:define-constant +standard-table-output-record-initargs+
+      '(x-spacing y-spacing multiple-columns
+        multiple-columns-x-spacing equalize-column-widths)
+    :test #'equal))
 
 (defun invoke-formatting-table
-    (stream continuation
-     &key x-spacing y-spacing
-       multiple-columns
-       multiple-columns-x-spacing
-       equalize-column-widths
-       (move-cursor t)
-       (record-type 'standard-table-output-record)
-       &allow-other-keys)
+    (stream continuation &rest initargs
+                         &key (record-type 'standard-table-output-record)
+                              (move-cursor t)
+                              . #.`(,@+standard-table-output-record-initargs+
+                                    &allow-other-keys))
```

Comment:
mention all arguments manually and for consistency define as a generic function (and add it to decls)

Reply:
mentioned later.